config/fcos/v1_6_exp/translate.go

```diff
@@ -144,11 +143,11 @@ func (c Config) processBootDevice(config *types.Config, ts *translate.Translatio
 		wantEFIPart = true
 	case *layout == "ppc64le":
 		wantPRePPart = true
-	case *layout == "s390x-zfcp":
+	case *layout == "s390x-zfcp" && wantLuksDevice:
```

Comment:
The intention is as per the design validation in github issue https://github.com/coreos/butane/issues/453#issuecomment-1652993830  by Benjamin. The concern is Sometimes user forgot to provide the device like dasda or sd in luks.device and tend to use the label. For that purpose i made a condition. So that User must provide the device corresponding to the layout.

I'll re-write with another logic as if the wantLuksDevice could not found it panics and crashes rather than just giving an error message like device is must for s390x layouts.